dataplane/vrf.cpp

```diff
@@ -210,6 +226,372 @@ VrfLpm4Linear::OneBlock* VrfLpm4Linear::Block(BlocksAllocator::Index index) cons
 	return reinterpret_cast<VrfLpm4Linear::OneBlock*>((index == BlocksAllocator::null_block ? nullptr : allocator_[index]));
 }
 
+/*
+ * Implementation VrfLpm4BinaryTree
+ */
+
+VrfLpm4BinaryTree::VrfLpm4BinaryTree(BlocksAllocator& allocator) :
+        allocator_(allocator)
+{
+	Init();
+}
+
+eResult VrfLpm4BinaryTree::Insert(stats_t& stats, tVrfId vrfId, const uint32_t& ipAddress, const uint8_t& mask, const uint32_t& valueId)
+{
+	if (mask > 32 || valueId & 0xFF000000)
+	{
+		YANET_LOG_DEBUG("invalid prefix or value\n");
+		return eResult::invalidArguments;
+	}
+	if (vrfId >= YANET_RIB_VRF_MAX_NUMBER)
```

Comment:
Fix the same issues as with Linear, please.
Like avoidance of reference field member, `emplace` instead of push with braces initialization: `push({...})`, should throw in constructor if the size allocated is not enough for `YANET_RIB_VRF_MAX_NUMBER`, otherwise loop like this
```cpp
	for (tVrfId vrfId = 0; vrfId < YANET_RIB_VRF_MAX_NUMBER; vrfId++)
	{
		OneBlock* block = Block(vrfId);
```
will segfault on `Block` when we'll call `allocator_[index]`

And also an observation that we can cache the blocks by a pair ipAddress + mask to avoid looking them up each time